xwiki-platform-core/xwiki-platform-edit/xwiki-platform-edit-ui/src/main/resources/XWiki/InplaceEditing.xml

```diff
@@ -873,45 +879,33 @@ define('editInPlace', [
       return this._getActionButtons().find(selector)[0];
     },
     serialize: function() {
-      var extra = this._getActionButtons().find(':input').serializeArray().reduce(function(extra, entry) {
-        var value = extra[entry.name] || [];
-        value.push(entry.value);
-        extra[entry.name] = value;
-        return extra;
-      }, {});
+      var formData = new FormData();
+      addToFormData(formData, this._getActionButtons().find(':input'));
       // retrieve all input fields listing the temporary uploaded files.
-      var uploadedFiles = $('#xwikicontent').nextAll('input[name="uploadedFiles"]').serializeArray().reduce(function(extra, entry) {
-        var value = extra[entry.name] || [];
-        value.push(entry.value);
-        extra[entry.name] = value;
-        return extra;
-      }, {});
       var xwikiDocument = this._getActionButtons().data('xwikiDocument');
-      var formData = {
-        title: xwikiDocument.rawTitle,
-        language: xwikiDocument.getRealLocale(),
-        isNew: xwikiDocument.isNew
-      };
-      if (xwikiDocument.content != xwikiDocument.originalDocument.content) {
+      formData.append('title', xwikiDocument.rawTitle);
+      formData.append('language', xwikiDocument.getRealLocale());
+      formData.append('isNew', xwikiDocument.isNew);
+
+      if (xwikiDocument.content !== xwikiDocument.originalDocument.content) {
         // Submit the raw (source) content. No syntax conversion is needed in this case.
-        formData.content = xwikiDocument.content;
+        formData.append('content', xwikiDocument.content);
       } else {
         // Submit the rendered content (HTML), but make sure it is converted to the document syntax on the server.
-        $.extend(formData, {
-          content: xwikiDocument.renderedContent,
-          RequiresHTMLConversion: 'content',
-          content_syntax: xwikiDocument.syntax
-        });
+        formData.append('content', xwikiDocument.renderedContent);
+        formData.append('RequiresHTMLConversion', 'content');
+        formData.append('content_syntax', xwikiDocument.syntax);
```

Comment:
you could have used 
```
addToFormData(formData, {
  'content': xwikiDocument.renderedContent, 
  'content_syntax': ... 
  ...
})
```
here no?

Reply:
No, see my comments below. This needs to overwrite the already set data (coming from ``editactions.vm``).